### PR TITLE
test/e2e/upgrade: Use multi-node cluster for upgrade testing

### PR DESCRIPTION
**Pull request by sunjayBhatia**
Now we are testing routability *during* an upgrade, we need to have a
more HA deployment of Envoy to test against. This change ensures we have
2 nodes in our kind cluster so the Envoy daemonset is not completely
unavailable during an upgrade.

Kind setup script optionally installs MetalLB in the kind cluster to provide load balancing between nodes

Updates: #3679

We can potentially use this setup for the regular e2e tests, this is just a fix for the upgrade tests currently

### CODE REVIEW

**Comment by sunjayBhatia** [comment]
hm, (hopefully) separate flake in the e2e tests that use port forwarding

**Comment by sunjayBhatia** [comment]
> hm, (hopefully) separate flake in the e2e tests that use port forwarding

maybe not a flake, possible influenced by change to `WaitForEnvoyDaemonSetUpdated`

File: test/scripts/make-kind-cluster.sh [hunk 1]
```diff
@@ -77,6 +82,35 @@ do
     kind::cluster::load "$image"
 done
 
+if [[ "${MULTINODE_CLUSTER}" == "true" ]]; then
+    # Install metallb.
+    ${KUBECTL} apply -f https://raw.githubusercontent.com/metallb/metallb/master/manifests/namespace.yaml
+    if ! kubectl get secret -n metallb-system memberlist; then
+        ${KUBECTL} create secret generic -n metallb-system memberlist --from-literal=secretkey="$(openssl rand -base64 128)"
+    fi
+    ${KUBECTL} apply -f https://raw.githubusercontent.com/metallb/metallb/master/manifests/metallb.yaml
```
Comment (skriss) [inline test/scripts/make-kind-cluster.sh:91]: probably makes sense to pin a tagged version of metallb, WDYT?

**Review by skriss — CHANGES_REQUESTED** [review]
Looks reasonable to me, 🤞 that the multi-node cluster doesn't bring GH Actions to its knees. Might be a good idea to try re-running it a bunch of times before we merge, to see how the reliability is.

File: test/scripts/make-kind-cluster.sh (hunk 1 above)
Comment (sunjayBhatia) [reply test/scripts/make-kind-cluster.sh:91]: good point!

**Review by stevesloka — APPROVED** [review]

**Review by skriss — APPROVED** [review]

**Pull request merged**